pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -737,8 +737,9 @@ func (nc *Controller) monitorNodeHealth(ctx context.Context) error {
 			pods, err := nc.getPodsAssignedToNode(node.Name)
 			if err != nil {
 				utilruntime.HandleError(fmt.Errorf("unable to list pods of node %v: %v", node.Name, err))
-				if currentReadyCondition.Status != v1.ConditionTrue && observedReadyCondition.Status == v1.ConditionTrue {
-					// If error happened during node status transition (Ready -> NotReady)
+				if (currentReadyCondition.Status != v1.ConditionTrue && observedReadyCondition.Status == v1.ConditionTrue) ||
+					(currentReadyCondition.Status == v1.ConditionUnknown && observedReadyCondition.Status != currentReadyCondition.Status) {
```

Comment:
maybe `currentReadyCondition.Status == v1.ConditionUnknown && observedReadyCondition.Status == v1.ConditionFalse` is enough

as `currentReadyCondition.Status == v1.ConditionUnknown && observedReadyCondition.Status == v1.ConditionTrue` is included by the first match

Reply:
Before describing the conditions, the background is as follows:

- observed: Represents the `Node Ready Conditions`.
- current: Calculated based on heartbeat time and `Node Ready Conditions`, and when observed and current are differ, `nodeReadyConditions` will be updated by current.


The scenario here is: `tryUpdateNodeHealth` succeeds, `getPodsAssignedToNode` fails, at this time, the `nodeReadyConditions` are already in an Unknown state. To ensure that MarkPodNotReady is executed next time, this matching condition is added.


Here, when `getPodsAssignedToNode` fail, retryBit will be set, which will trigger MarkPodNotReady in the subsequent code.

Reply:
o , sorry for misleading, what I mean is  `(currentReadyCondition.Status != v1.ConditionTrue && observedReadyCondition.Status == v1.ConditionTrue) ||
					(currentReadyCondition.Status == v1.ConditionUnknown && observedReadyCondition.Status == v1.ConditionFalse)` is enough

not `(currentReadyCondition.Status != v1.ConditionTrue && observedReadyCondition.Status == v1.ConditionTrue) ||
					(currentReadyCondition.Status == v1.ConditionUnknown && observedReadyCondition.Status != currentReadyCondition.Status)`

Reply:
uhhh...  sure, it is work